query.go

```diff
@@ -194,6 +195,24 @@ func (s *Selector) Do(ctx context.Context) error {
 		if nodes == nil || err != nil {
 			continue
 		}
+
+		if s.filter != nil {
```

Comment:
Hmm, we should think about moving this before "wait" in the future, but that would require an API-breaking refactor. because, ideally, the steps would be:

```
nodes, err := s.findNodes(...)
if nodes == nil || err != nil { continue }

if s.filter != nil { ... }
if s.wait != nil { ... }
if s.after != nil { ... }
```

This is useful, because imagine if you wanted to do `FilterVisible, NodeEnabled` to find all text inputs which are visible, then wait for them to be enabled (so that you can type into them). The order here is very important; with the current order, if there is a non-visible element which is disabled, you would first wait for it to be enabled, so you would block forever before the filter does its thing.

We don't have to do this change here, but I would add a TODO.